### PR TITLE
Remove support for setext headings

### PR DESCRIPTION
**Pull request by dgileadi**
Per @WalterBright's [comment](https://github.com/dlang/dmd/pull/8952#issuecomment-439338688) on the Markdown headings PR, this PR is to remove support for setext headings in Ddoc.

Please note that per [my comment](https://github.com/dlang/dmd/pull/8952#issuecomment-438867860) in the same PR I disagree with this PR to remove setext headings—I believe keeping them offers wider compatibility with Ddoc alternatives.

### CODE REVIEW

**Comment by thewilsonator** [comment]
> I disagree with this PR to remove setext headings—I believe keeping them offers wider compatibility with Ddoc alternatives

So do I.

**Review by thewilsonator — DISMISSED** [review]
To boot, we should not claim to support markdown without stet headings.

**Review by WalterBright — APPROVED** [review]

**Comment by WalterBright** [comment]
After this is merged, it should also go into the Phantom Zone so that if it because clear it needs to be added, the implementation work is already done.

**Comment by thewilsonator** [comment]
> so that if it because clear it needs

you broke my parser.

**Comment by WalterBright** [comment]
"becomes" - I have no idea how that got so garbled!

**Comment by thewilsonator** [comment]
Yay for autocorrect!

**Review by WalterBright — APPROVED** [review]

**Comment by thewilsonator** [comment]
Removing auto-merge, this is clogging CIs. 

I meant to talk about this at the meeting but forgot sorry. If you _positively do not want_ to support this feature, as opposed to merely think it might provide value later then you should dismiss my review and readd auto merge, if not please close this.

However, this is not one of those things that will "[become] clear it needs to be added", its one of those things that are going to _seriously confuse_ people trying to use it because we claim to support markdown but only support whatever the flavour with the `#`s is, not setext. IMO Setext is nicer to read in plaintext i.e. when reading code. I would consider this a seriously incomplete implementation with it.

**Comment by WalterBright** [comment]
I can never tell if sempahoreci failures are actual problems or not.

**Comment by WalterBright** [comment]
> But, I definitely think we should support all Markdown syntax or not claim it's Markdown.

We can't support it 100% anyway, so that point is moot.

The issue that matters is does it make it nice & easy to write documentation.

**Comment by thewilsonator** [comment]
> We can't support it 100% anyway,

Silly question, why?

**Comment by dgileadi** [comment]
For reference, here's a list of things that I didn't implement in Ddoc Markdown due to incompatibilities:

* Whitespace-indented code blocks, because text in documentation comments almost never starts at the line start
* The `---` form of setext headings, because of conflicts with code blocks
* Likewise the `---` form of horizontal breaks (although `- - -` will be supported)
* The use of `_` for emphasis
* The `1)` form of numbered lists, because of conflicts with macros
* Embedded HTML support follows Ddoc's existing rules, rather than CommonMark rules
* Paragraphs are separated by `<br><br>` instead of being wrapped in `<p>`'s, with the result of not collapsing consecutive whitespace

Again, I am a proponent of supporting as much standard Markdown as we can, for two reasons:

1. People familiar with Markdown will be able to type what they're used to and it will work as expected, modulo the above items
2. Compatibility with Markdown support in other Ddoc implementation that support Markdown features like [adrdox](http://dpldocs.info/experimental-docs/adrdox.syntax.html) and [harbored-mod](https://github.com/dlang-community/harbored-mod).

**Pull request merged**
